app/code/community/Algolia/Algoliasearch/Helper/Entity/Producthelper.php

```diff
@@ -1041,6 +1043,44 @@ public function getObject(Mage_Catalog_Model_Product $product)
         return $customData;
     }
 
+    /**
+     * Returns all parent product IDs, e.g. when simple product is part of configurable or bundle
+     *
+     * @param array $productIds
+     * @return array
+     */
+    public function getParentProductIds(array $productIds)
+    {
+        $parentIds = array();
+        foreach ($this->getCompositeTypes() as $typeInstance) {
+            $parentIds = array_merge($parentIds, $typeInstance->getParentIdsByChild($productIds));
+        }
+        return $parentIds;
```

Comment:
Can you leave a blank line before the `return`? It's a bit more readable :)

Reply:
You got it!